drivers/serial/uart_nrfx_uarte.c

```diff
@@ -8,21 +8,24 @@
  * @brief Driver for Nordic Semiconductor nRF UARTE
  */
 
+#include <zephyr/kernel.h>
 #include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/pinctrl.h>
 #include <zephyr/pm/device.h>
-#include <hal/nrf_uarte.h>
-#include <nrfx_timer.h>
 #include <zephyr/sys/util.h>
-#include <zephyr/kernel.h>
-#include <soc.h>
-#include <helpers/nrfx_gppi.h>
 #include <zephyr/linker/devicetree_regions.h>
 #include <zephyr/irq.h>
-
+#include <zephyr/cache.h>
```

Comment:
Is this header actually needed? And `soc.h`?
And what was the key in header inclusion rearrangement? It still looks a bit untidy. Maybe using alphabetical order and splitting Zephyr and nrfx ones could help?

Reply:
Removed changes in headers.